blueocean-pipeline-editor/src/main/js/EditorPage.jsx

```diff
@@ -82,6 +82,12 @@ class SaveDialog extends React.Component {
         this.setState({ saving: false, errorMessage });
     }
 
+    processBranchName(event) {
+        const cleanBranchName = event.target.value.replace(/\s/g, "");
+
+        this.setState({branch: this.branchOptions[1].branch = cleanBranchName});
```

Comment:
This code looks susceptible to a NPE.

Reply:
Also assignment inside of a `setState` like that is not something I've seen us do before, should probably be pulled out to its own line with null / index check.

Reply:
Sorry, I see the original code was actually written that way, but I would split it up anyways.